core/include/moveit/task_constructor/stages/generate_pose.h

```diff
@@ -57,10 +59,67 @@ class GeneratePose : public MonitoringGenerator
 
 	void setPose(const geometry_msgs::PoseStamped pose) { setProperty("pose", std::move(pose)); }
 
+	/*
+	 * Function specification for pose dimension samplers.
+	 * The input paramter is the target_pose value used as seed, the returned value is the sampling result.
+	 * */
+	typedef std::function<double(double)> PoseDimensionSampler;
+	enum PoseDimension
+	{
+		X,
+		Y,
+		Z,
+		ROLL,
+		PITCH,
+		YAW
+	};
+
+	/*
+	 * Configure a RandomNumberDistribution (see https://en.cppreference.com/w/cpp/numeric/random) sampler for a
+	 * PoseDimension (X/Y/Z/ROLL/PITCH/YAW) for randomizing the pose.
+	 * The distribution_param is applied to the specified distribution method while the target pose value is used as
+	 * seed.
+	 * Currently supported distributions are:
+	 * * std::uniform_real_distrubtion: distribution_param specifies the full value range around the target_pose value
+	 * * std::normal_distribution: distribution_param is used as stddev, target_pose value is mean
+	 *
+	 * Other distributions can be used by setting the PoseDimensionSampler directly using the function overload.
+	 */
+	template <template <class Realtype = double> class RandomNumberDistribution>
+	void sampleDimension(const PoseDimension pose_dimension, double distribution_param) {
```

Comment:
Would it make sense to make `distribution_param` a [parameter pack](https://en.cppreference.com/w/cpp/language/parameter_pack) to enable support for distributions with more than one parameter?

Reply:
Ignore my comment. This is fine as-is.